packages/plugins/users-permissions/server/controllers/validation/user.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const { yup } = require('@strapi/utils');
+const { YupValidationError } = require('@strapi/utils').errors;
+
+const validateSchema = schema => async (body, errorMessage) => {
+  try {
+    await schema.validate(body, { strict: true, abortEarly: false });
+  } catch (e) {
+    throw new YupValidationError(e, errorMessage);
+  }
+};
+
+const deleteRoleSchema = yup.object().shape({
+  role: yup.strapiID().required(),
+});
+
+const createUserBodySchema = yup
+  .object()
+  .shape({
+    email: yup
+      .string()
+      .email()
+      .required(),
+    username: yup
+      .string()
+      .min(1)
+      .required(),
+    password: yup
```

Comment:
while at it we can enforce some better default I think

Reply:
I would prefer to focus on rest. Is it ok for you if we make this change later?